Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -0,0 +1,306 @@
+                // asset safe folder (typically the project root)
+                if (resultCode == QMessageBox::Retry)
+                {
+                    OnOpenButtonClicked();
+                }
+
+                // Make sure our error dialog gets deleted after it is dismissed.
+                errorDialog->deleteLater();
+            });
+
+            errorDialog->open();
+        }
+    }
+
+    void PropertyFilePathCtrl::OnClearButtonClicked()
+    {
+        m_currentFilePath.clear();
+        m_browseEdit->lineEdit()->clearFocus();
+
+        Q_EMIT FilePathChanged();
+    }
+
+    void PropertyFilePathHandler::ConsumeAttribute(PropertyFilePathCtrl* GUI, AZ::u32 attrib, PropertyAttributeReader* attrValue, [[maybe_unused]] const char* debugName)
+    {
+        if (attrib == AZ::Edit::Attributes::SourceAssetFilterPattern)
+        {
+            AZStd::string filter;
+            if (attrValue->Read<AZStd::string>(filter))
+            {
+                GUI->SetFilter(filter.c_str());
+            }
+        }
+        else if (attrib == AZ::Edit::Attributes::DefaultAsset)
+        {
+            AZStd::string defaultAsset;
+            if (attrValue->Read<AZStd::string>(defaultAsset))
+            {
+                GUI->SetDefaultFileName(defaultAsset);
+            }
+        }
+    }
+
+    AZ::u32 PropertyFilePathHandler::GetHandlerName() const
+    {
+        return AZ_CRC_CE("FilePath");
+    }
+
+    bool PropertyFilePathHandler::IsDefaultHandler() const
+    {
+        return true;
+    }
+
+    QWidget* PropertyFilePathHandler::CreateGUI(QWidget* parent)
+    {
+        PropertyFilePathCtrl* newCtrl = aznew PropertyFilePathCtrl(parent);
+
+        QObject::connect(newCtrl, &PropertyFilePathCtrl::FilePathChanged, this, [newCtrl]()
+        {
+            PropertyEditorGUIMessages::Bus::Broadcast(&PropertyEditorGUIMessages::RequestWrite, newCtrl);
+            PropertyEditorGUIMessages::Bus::Broadcast(&PropertyEditorGUIMessages::Bus::Handler::OnEditingFinished, newCtrl);
+        });
+
+        return newCtrl;
+    }
+
+    void PropertyFilePathHandler::WriteGUIValuesIntoProperty(size_t index, PropertyFilePathCtrl* GUI, property_t& instance, InstanceDataNode* node)
+    {
+        AZ_UNUSED(index);
+        AZ_UNUSED(node);
+
+        AZ::IO::Path filePath = GUI->GetFilePath();
+        instance = static_cast<property_t>(filePath);
+    }
+
+    bool PropertyFilePathHandler::ReadValuesIntoGUI(size_t index, PropertyFilePathCtrl* GUI, const property_t& instance, InstanceDataNode* node)
+    {
+        AZ_UNUSED(index);
+        AZ_UNUSED(node);
+
+        GUI->blockSignals(true);
+
+        GUI->SetFilePath(instance);
+
+        GUI->blockSignals(false);
+
+        return false;
+    }
+
+    void RegisterFilePathHandler()
+    {
+        PropertyTypeRegistrationMessages::Bus::Broadcast(
+            &PropertyTypeRegistrationMessages::RegisterPropertyType, aznew PropertyFilePathHandler());
+    }
+}
```

Comment:
nit: Missing closing namespace (you could try running clang-format on this new file 🙂 it'll do it for you! 😅)